src/consumer.rs

```diff
@@ -184,9 +202,57 @@ pub async fn from_substrait_rel(ctx: &mut SessionContext, rel: &Rel) -> Result<A
                 ))
             }
         }
+        Some(RelType::Aggregate(agg)) => {
+            if let Some(input) = agg.input.as_ref() {
+                let input = from_substrait_rel(ctx, input, extensions).await?;
+                let mut group_expr = vec![];
+                let mut aggr_expr = vec![];
+
+                let groupings = match agg.groupings.len() {
+                    1 => { Ok(&agg.groupings[0]) },
+                    _ => {
+                        Err(DataFusionError::NotImplemented(
+                            "Aggregate with multiple grouping sets is not supported".to_string(),
+                        ))
+                    }
+                };
+
+                for e in &groupings.unwrap().grouping_expressions {
+                    let x = from_substrait_rex(&e, &input.schema(), extensions).await?;
+                    group_expr.push(x.as_ref().clone());
+                }
+
+                for m in &agg.measures {
+                    let filter = match &m.filter {
+                        Some(fil) => Some(Box::new(from_substrait_rex(fil, &input.schema(), extensions).await?.as_ref().clone())),
+                        None => None
+                    };
+                    let agg_func = match &m.measure {
+                        Some(f) => {
+                            let distinct = match f.invocation  {
+                                _ if f.invocation == AggregationInvocation::Distinct as i32 => true,
+                                _ if f.invocation == AggregationInvocation::All as i32 => false,
+                                _ => false
+                            };
+                            Ok(from_substrait_agg_func(&f, &input.schema(), extensions, filter, distinct).await?)
+                        },
+                        None => Err(DataFusionError::NotImplemented(
```

Comment:
Same comment as above ... we can probably `return Err` here and remove the need for the unwrap.